### DISCUSSION TITLE
ring-opacity doesn't work with hsl

### DISCUSSION THREAD

**Discussion by Zerebokep** [tailwindlabs/tailwindcss#9794] [Help] [answered]
**What version of Tailwind CSS are you using?**

For example: v3.2.2

**What build tool (or framework if it abstracts the build tool) are you using?**

nextjs 13

**What version of Node.js are you using?**

most recent docker img

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Describe your issue**

ring-opacity doesn't seem to work with hsl defined colors, e.g.

`black: 'hsl(var(--color-black) / 1)',`

using

`ring-black ring-opacity-5` does not change the opacity

`ring-black/5` works

**Comment by adamwathan** [comment, marked as answer]
Hey! Make sure to read the docs on [using CSS variables as colors](https://tailwindcss.com/docs/customizing-colors#using-css-variables) in Tailwind, it works if you follow the instructions:

https://play.tailwindcss.com/k0qYAv5EWf?file=config

**Reply by lord007tn** [reply]
same with rgb works but when using hsl and following docs doesn't seem to work
